aiida_testing/mock_code/_cli.py

```diff
@@ -27,23 +28,44 @@ def run() -> None:
     directory.
     """
     # Get environment variables
+    log_path = Path(os.environ[EnvKeys.LOG_FILE.value])
     label = os.environ[EnvKeys.LABEL.value]
     data_dir = os.environ[EnvKeys.DATA_DIR.value]
     executable_path = os.environ[EnvKeys.EXECUTABLE_PATH.value]
     ignore_files = os.environ[EnvKeys.IGNORE_FILES.value].split(':')
     ignore_paths = os.environ[EnvKeys.IGNORE_PATHS.value].split(':')
     regenerate_data = os.environ[EnvKeys.REGENERATE_DATA.value] == 'True'
+    fail_on_missing = os.environ[EnvKeys.FAIL_ON_MISSING.value] == 'True'
 
-    hash_digest = get_hash().hexdigest()
+    def _log(msg: str) -> None:
+        """Write a message to the log file."""
+        with open(log_path, 'a', encoding='utf8') as log_file:
+            log_file.write(f"{datetime.now()}:{label}: {msg}\n")
+
+    _log('Init mock code')
+
+    hash_digest = get_hash(_log).hexdigest()
 
     res_dir = Path(data_dir) / f"mock-{label}-{hash_digest}"
 
-    if regenerate_data and res_dir.exists():
-        shutil.rmtree(res_dir)
+    if res_dir.exists():
+        _log(f"Cached directory exists: {res_dir}")
+        if regenerate_data:
+            _log("Regenerating data")
+            shutil.rmtree(res_dir)
+    else:
+        _log(f"Cached directory does not exist: {res_dir}")
+
+        if fail_on_missing:
+            _log(f"ERROR: Failing on missing cached data: {res_dir}")
+            sys.exit("Missing cached data")
 
     if not res_dir.exists():
         if not executable_path:
-            sys.exit("No existing output, and no executable specified.")
+            _log(f"Failing on missing cached data: {res_dir}")
+            sys.exit("ERROR: No existing output, and no executable specified.")
```

Comment:
```suggestion
            sys.exit("No existing output, and no executable specified.")
```

Reply:
ah, this was meant to be in the log message not the exit one